.github/workflows/test.yml

```diff
@@ -27,19 +27,21 @@ jobs:
           path: "./check out"
           persist-credentials: false
 
-      - name: Set up Python
-        uses: actions/setup-python@e797f83bcb11b83ae66e0230d6156d7c80228e7c # v6.0.0
+      - name: Install uv
+        uses: astral-sh/setup-uv@b75a909f75acd358c2196fb9a5f1299a9a8868a4 # v6.7.0
         with:
           python-version: ${{ matrix.python-version }}
+          working-directory: './check out/'
+          enable-cache: true
 
       - name: Display Python version
         run: python -c "import sys; print(sys.version); print(sys.platform)"
 
-      - name: Install tox
-        run: pip3 install tox
+      - name: Setup project
+        run: uv sync --frozen --directory "./check out/"
 
-      - name: Run tox
-        run: tox -c 'check out' -- -W error
+      - name: Run tasks
+        run: uv run --frozen --directory "./check out/" poe all
```

Comment:
Could you not run "all" here to provide finer-grained results? And durations.

Reply:
This is close to what `tox` did, so I'll leave it for now. I think we should split indeed, but also reduce the matrix a bit. We don't need to run the linter task 15 times. Will follow-up!

Reply:
Submitted in:
- #858